service/service.go

```diff
@@ -303,31 +303,14 @@ func New(config Config) (*Service, error) {
 }
 
 func shouldCreateLegacyController(clients k8sclient.Interface, provider string) (bool, error) {
-	switch provider {
-	case "aws":
-		_, err := clients.G8sClient().ProviderV1alpha1().AWSConfigs("default").List(context.Background(), v1.ListOptions{})
-		if apierrors.IsNotFound(err) {
-			return false, nil
-		} else if err != nil {
-			return false, microerror.Mask(err)
-		}
-	case "azure":
-		_, err := clients.G8sClient().ProviderV1alpha1().AzureConfigs("default").List(context.Background(), v1.ListOptions{})
-		if apierrors.IsNotFound(err) {
-			return false, nil
-		} else if err != nil {
-			return false, microerror.Mask(err)
-		}
-	case "kvm":
-		_, err := clients.G8sClient().ProviderV1alpha1().KVMConfigs("default").List(context.Background(), v1.ListOptions{})
-		if apierrors.IsNotFound(err) {
-			return false, nil
-		} else if err != nil {
-			return false, microerror.Mask(err)
-		}
-	case "vmware":
-		return false, nil
+	name := fmt.Sprintf("%sconfigs.v1alpha1.provider.giantswarm.io", provider)
+
+	var crd apiextensionsv1.CustomResourceDefinition
+	err := clients.CtrlClient().Get(context.Background(), client.ObjectKey{Name: name}, &crd)
+	if err != nil {
+		return false, microerror.Mask(err)
```

Comment:
Since `k8sclient.G8sClient()` is gone (see https://github.com/giantswarm/apiextensions/pull/845), I rather move to checking if CRDs are installed instead of trying to generate the correct client for legacy configs (e.g. AWSConfig).

Reply:
<3